crates/hotshot/src/lib.rs

```diff
@@ -802,59 +701,60 @@ impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusSharedApi<TYPES, I>
     }
 }
 
-#[async_trait]
-impl<TYPES: NodeType, I: NodeImplementation<TYPES>> ConsensusApi<TYPES, I>
-    for HotShotConsensusApi<TYPES, I>
-{
-    async fn send_direct_message(
+impl<TYPES: NodeType, I: NodeImplementation<TYPES>> HotShotConsensusApi<TYPES, I> {
+    /// Send a direct message to the given recipient
+    ///
+    /// # Errors
+    /// - [`NetworkError`] if sending the message failed
+    pub async fn send_direct_message(
         &self,
         recipient: TYPES::SignatureKey,
         message: SequencingMessage<TYPES>,
     ) -> std::result::Result<(), NetworkError> {
         let inner = self.inner.clone();
         debug!(?message, ?recipient, "send_direct_message");
-        async_spawn_local(async move {
```

Comment:
I don't think we need `async_spawn_local` here anymore, do we?

Reply:
Hmmmmm, good question

I think we still want it to be non-blocking (because some of the downstream futures like DMing on libp2p take kind of long), but not through `async_spawn_local`. I think we should only use that for futures that are not `Send`; we can probably get away with `async_spawn` for this.

However I don't think this function's caller is actually used anywhere upstream, maybe we can remove it entirely 😃

Reply:
My reasoning was if the caller wants this not to be blocking, they can just do the same thing on their side. Also, this has an advantage of not just dropping any errors that might arise.

Reply:
> However I don't think this function's caller is actually used anywhere upstream, maybe we can remove it entirely 😃

+1, we could probably remove more functions inside this `impl`.